docs/fee_design.md

```diff
@@ -0,0 +1,173 @@
+
+As a result, fee rules update is done by changing the logic of how the child-chain service would accept/reject an incoming transaction and how watcher/wallet update the logic to follow the new fee rule when generating transaction.
+
+
+### Fee for new transaction type
+In order to allow fees for a new transaction type, one would need to define how the fee is collected in the transaction type. We would flavor the fee to be always collected in an implicit way as Payment transaction has already decided to collect fee implicitly. To collect fees explicitly, the exit game contract of such transaction type might need to make sure the fee exit via standard fee exit mechanism instead of directly exit with the explicit fee transaction output.
+
+Once the way to collect fee is designed and defined, whenever there is a new transaction coming in, child-chain and watcher should add the fee amount to the storage that records sum of fee.
+
+### Fee rule change within a transaction type
+Child chain service (the operator) needs to provide the fee rules to its clients. Clients can pass in the essential information such as address, token, transaction type ...etc, and then the child chain can accept transactions that pay sufficient fees.
+
+This gives the operator the ability to update fees in a traditional SaaS way. The operator can adjust fee rules at anytime with flexible control. It can be upgraded with a feature flag or A/B tests.
+
+
+## Chosen Design: Generate fee transaction to summarize the fees of each block
+
+The design proposes that the child chain service automatically generates fee transactions at the end of each block. And the child chain and watcher perform the fee transaction verification on the block level. A valid block, apart from the existing transaction verification logic exits if:
+1. All fee transactions are included after all other types of transactions in a block. 
+1. A fee transaction's output is the sum of fees paid in a particular token by all transactions in a block.
+1. There must be at most one fee transaction for a token per block.
+1. There can be at most one fee transaction type per block. (PS. as we extends the Plasma Framework, there can potentially be multiple transaction types that all counts as "fee tx type")
+
+Since the transactions are automatically generated from the child chain service, there is no need to do an authentication check. Inclusion of fee transaction is optional, however fees collected in omitted blocks are lost to claim.
+
+### Transaction design
+The transaction would be of the following format:
+
+```
```

Comment:
The Generic tx goes a bit off aftertastes latest change to doc. It has an extra field “metaData” now, can be any bytes32 as it’s our protocol.

Can you help add the field?

Reply:
Added :)